src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1033,17 +1766,57 @@ ExtractInsertPartitionValue(Query *query, Var *partitionColumn)
 
 /* RouterSelectTask builds a Task to represent a single shard select query */
 static Task *
-RouterSelectTask(Query *originalQuery, Query *query,
-				 RelationRestrictionContext *restrictionContext,
+RouterSelectTask(Query *originalQuery, RelationRestrictionContext *restrictionContext,
 				 List **placementList)
 {
 	Task *task = NULL;
-	List *prunedRelationShardList = TargetShardIntervalsForSelect(query,
-																  restrictionContext);
+	bool queryRoutable = false;
 	StringInfo queryString = makeStringInfo();
-	uint64 shardId = INVALID_SHARD_ID;
 	bool upsertQuery = false;
-	CmdType commandType PG_USED_FOR_ASSERTS_ONLY = query->commandType;
+	uint64 shardId = INVALID_SHARD_ID;
+	List *selectShardList = NIL;
+
+	queryRoutable = RouterSelectQuery(originalQuery, restrictionContext,
+									  placementList, &shardId, &selectShardList);
+
+
+	if (!queryRoutable)
+	{
+		return NULL;
+	}
+
+	pg_get_query_def(originalQuery, queryString);
+
+	task = CitusMakeNode(Task);
+	task->jobId = INVALID_JOB_ID;
+	task->taskId = INVALID_TASK_ID;
+	task->taskType = ROUTER_TASK;
+	task->queryString = queryString->data;
+	task->anchorShardId = shardId;
+	task->dependedTaskList = NIL;
+	task->upsertQuery = upsertQuery;
+
+	return task;
+}
+
+
+/*
+ * RouterSelectQuery returns true if the input query can be pushed down to the
+ * worker node as it is. Otherwise, the function returns false.
+ *
+ * On return true, all RTEs have been updated to point to the relevant nodes in
```

Comment:
It's "relative shards" not "relative nodes", right? Because with RF > 1, they each can be on more than one worker.